ui-admin/src/study/participants/enrolleeView/EnrolleeProfile.tsx

```diff
@@ -1,84 +1,298 @@
-      <label className="form-label">
-        Postal code: <input className="form-control" type="text" readOnly={true} value={mailingAddress.postalCode}/>
-      </label>
+/**
+ * Read only view of the profile.
+ */
+export function ReadOnlyProfile(
+  { profile }: {
+    profile: Profile
+  }
+) {
+  const mailingAddress = profile.mailingAddress
+  return <>
+    <ReadOnlyRow title={'Name'} values={[`${profile.givenName} ${profile.familyName}`]}/>
+    <ReadOnlyRow title={'Birthdate'} values={[dateToDefaultString(profile.birthDate)]}/>
+    <ReadOnlyRow title={'Primary Address'} values={[
+      mailingAddress.street1,
+      mailingAddress.street2,
+      `${mailingAddress.city}, ${mailingAddress.state} ${mailingAddress.postalCode}`
+    ]}/>
+    <ReadOnlyRow title={'Email'} values={[profile.contactEmail]}/>
+    <ReadOnlyRow title={'Phone'} values={[profile.phoneNumber]}/>
+    <ReadOnlyRow title={'Notifications'} values={[profile.doNotEmail ? 'On' : 'Off']}/>
+    <ReadOnlyRow title={'Do Not Solicit'} values={[profile.doNotEmailSolicit ? 'On' : 'Off']}/>
+  </>
+}
+
+/**
+ * Editable profile, providing the update to `setProfile` on each change.
+ */
+export function EditableProfile(
+  { profile, setProfile }: {
+    profile: Profile,
+    setProfile: (value: React.SetStateAction<Profile>) => void
+  }
+) {
+  const onFieldChange = (field: string, value: string | boolean) => {
+    setProfile((oldVal: Profile) => {
+      return {
+        ...oldVal,
+        [field]: value
+      }
+    })
+  }
+
+  const onDateFieldChange = (field: string, date: Date | null) => {
+    const asJavaLocalDate: number[] | undefined = date ? jsDateToJavaLocalDate(date) : undefined
+
+    setProfile((oldVal: Profile) => {
+      return {
+        ...oldVal,
+        [field]: asJavaLocalDate
+      }
+    })
+  }
+
+  const onMailingAddressFieldChange = (field: string, value: string) => {
+    setProfile((oldVal: Profile) => {
+      return {
+        ...oldVal,
+        mailingAddress: {
+          ...oldVal.mailingAddress,
+          [field]: value
+        }
+      }
+    })
+  }
+
+  return <>
+    <FormRow title={'Name'}>
+      <div className='row'>
+        <div className='col'>
+          <input className="form-control" type="text" value={profile.givenName || ''}
+            placeholder={'Given Name'}
+            onChange={e => onFieldChange('givenName', e.target.value)}/>
+        </div>
+        <div className='col'>
+          <input className="form-control" type="text" value={profile.familyName || ''}
+            placeholder={'Family Name'}
+            onChange={e => onFieldChange('familyName', e.target.value)}/>
+        </div>
+      </div>
+    </FormRow>
+    <FormRow title={'Birthdate'}>
+      <div className='row'>
+        <div className="col">
+          <input className="form-control" type="date"
+            value={javaLocalDateToJsDate(profile.birthDate)?.toISOString().split('T')[0] || ''}
```

Comment:
Testing note: I suspect there's an issue converting back and forth between the different date formats. The initial value is correct, but when I use my keyboard to type in a new date, things start to get wonky once I start typing the year. I.e. typing 06/11/1993 is converted into 06/11/1903. The date selector dropdown works as expected.

Reply:
oooo good catch